### DISCUSSION TITLE
USB timeout error when flashing Jetson Orin NX device

### DISCUSSION THREAD

**Discussion by jordankrp** [kairos-io/kairos#2842] [Q&A]
This is a known [issue](https://forums.developer.nvidia.com/t/error-might-be-timeout-in-usb-write/254513/3) from NVIDIA and was wondering if anyone here has experienced the same. After trying both Ubuntu 18 and 20 as well as kernel version 5.10 and 5.15, it seems that the following error appears consistently when flashing Kairos to an NVIDIA Jetson Orin NX 16GB module:
```
[   6.0974 ] tegrarcm_v2 --chip 0x23 0 --pollbl --download bct_mem mem_rcm_sigheader.bct.encrypt --download blob blob.bin
[   6.0981 ] BL: version 0.32.0.0-t234-54845784-57325615 last_boot_error: 0
[   6.1269 ] Sending bct_mem
[   6.1717 ] Sending blob
[   6.2213 ] ERROR: might be timeout in USB write.
Error: Return value 3
Command tegrarcm_v2 --chip 0x23 0 --pollbl --download bct_mem mem_rcm_sigheader.bct.encrypt --download blob blob.bin
Failed flashing t186ref.
```
I also tried with different USB cables and ports, as well as disabling USB auto-suspend on my host. Any insight on this would be greatly appreciated.

**Comment by kencosmos** [comment]
I've encounted same problem

**Comment by mudler** [comment]
I was getting the same. The only solution was to find another box to do the flashing, neither VM or host usb redirection worked for me. Sadly at the end I could only make it working while using an Ubuntu-based distro.

**Comment by jimmykarily** [comment]
See also here: https://github.com/kairos-io/kairos/discussions/3648#discussioncomment-14727344
and here: https://kairos.io/docs/installation/nvidia_agx_orin/#usb-timeout-error)
